.github/workflows/metabox.yaml

```diff
@@ -10,16 +10,19 @@ jobs:
   metabox_run_required:
     runs-on: ubuntu-latest
     name: Check for changes in metabox and checkbox-ng dirs
+    outputs:
+      required_run: ${{ steps.check_diff.outputs.required_run }}
     steps:
       - uses: actions/checkout@v3
         with:
           fetch-depth: 0
       - name: Use git diff to see if there are any changes in the metabox and checkbox-ng directories
         id: check_diff
-        continue-on-error: true
         run: |
-          git diff --exit-code HEAD origin/main -- checkbox-ng metabox
-          if [[ $? -eq 0 ]]
+          # catch the return code in subshell to avoid github stopping the job 
+          # due to it being non-0, non-0 means: "found some diff"
+          SHOULD=$(git diff --exit-code HEAD origin/main -- checkbox-ng metabox 1>/dev/null 2>/dev/null; echo $?)
```

Comment:
This introduces a complexity on top of the previous bad choice.
Why not do something simple like `DIFF_LENGTH=git diff (...) |wc -n`